judge/create_cgroups.in

```diff
@@ -9,6 +9,14 @@
 JUDGEHOSTUSER=@DOMJUDGE_USER@
 CGROUPBASE="/sys/fs/cgroup"
 
+# We do not need to do any of this with cgroup v2.
+fs_type=$(awk '$2 == "/sys/fs/cgroup" {print $3}' /proc/mounts)
+if [ "$fs_type" = "cgroup2" ]; then
+    echo "+memory" >> /sys/fs/cgroup/cgroup.subtree_control
+    echo "+cpuset" >> /sys/fs/cgroup/cgroup.subtree_control
+    exit 0
+fi
+
 cgroup_error_and_usage () {
     echo "$1" >&2
     echo "To fix this, please make the following changes:
```

Comment:
In the first item below, we can probably remove the part about "modern distros" here now, like [here](https://github.com/DOMjudge/domjudge/pull/2588/files#diff-5102f182c3cec19492d278a25e12ed5771d066183cb3eeab004a9d6001130f3cL182) :smile:

The first sentence in this item (about adding kernel parameters) should probably stay.

Reply:
Done